hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/HBackupFileSystem.java

```diff
@@ -109,6 +120,16 @@ private static Path getManifestPath(Configuration conf, Path backupRootPath, Str
     return manifestPath;
   }
 
+  public static Path getRootDirFromBackupPath(Path backupPath, String backupId) {
+    if (backupPath.getName().equals(BackupManifest.MANIFEST_FILE_NAME)) {
+      backupPath = backupPath.getParent();
+    }
+    Preconditions.checkArgument(backupPath.getName().equals(backupId),
+      String.format("Backup path %s must end in backupId %s", backupPath, backupId));
+    return new Path(
```

Comment:
I'd simplify this to `return backupPath.getParent()`

Reply:
Good idea